vuepress/docs/tutorials/micro-frontends/communication.md

```diff
@@ -0,0 +1,506 @@
+        <button onClick={() => onGreet(name)}>Say hello!</button>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+### Test Event Dispatcher
+
+In the JavaScript console of your browser, enter:
+
+``` js
+window.addEventListener('greeting', (evt) => console.log('Hello', evt.detail.name))
+```
+
+Write something in the text field. Click the "Say hello!" button and take a look at the JS console. It will show the event message.
+
+::: tip Congratulations!
+You’ve now published a custom event.
+:::
+
+## Subscriber
+
+Next, let’s create the subscriber.
+
+``` bash
+npx create-react-app subscriber-widget --use-npm
+```
+
+Start the app.
+
+``` bash
+cd subscriber-widget
+```
+
+``` bash
+npm start
+```
+
+### Add Event Listener
+
+Add a new file `subscriber-widget/src/SubscriberWidgetElement.js`.
+
+``` js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+const EVENTS = {
+  greeting: 'greeting',
+};
+
+class SubscriberWidgetElement extends HTMLElement {
+
+  constructor() {
+    super();
+    this.name = null;
+    this.subscribeToWidgetEvent(EVENTS.greeting, (evt) => this.onGreeting(evt.detail.name));
+  }
+
+  connectedCallback() {
+    this.mountPoint = document.createElement('div');
+    this.appendChild(this.mountPoint);
+    this.render();
+  }
+
+  subscribeToWidgetEvent(eventType, eventHandler) {
+    window.addEventListener(eventType, eventHandler);
+  }
+
+  onGreeting(name) {
+    this.name = name;
+    this.render();
+  }
+
+  render() {
+    ReactDOM.render(<App name={this.name} />, this.mountPoint);
+  }
+}
+
+customElements.define('subscriber-widget', SubscriberWidgetElement);
+
+export default SubscriberWidgetElement;
+```
+
+### Import Custom Element
+
+Update `subscriber-widget/src/index.js`.
+
+``` js
+import './index.css';
+import './SubscriberWidgetElement';
+```
+
+#### Test Micro Frontend
+
+Update `subscriber-widget/public/index.html`, and view it in the browser.
+
+``` html
+  <body>
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <subscriber-widget>
+    ...
+  </body>
+```
+
+### Display Custom Event
+
+Update `subscriber-widget/src/App.js`.
+
+``` js
+import React from 'react';
+import './App.css';
+
+function App({ name }) {
+  return name ? (<h2>Just got a greeting from {name}</h2>)
+    : (<h2>Waiting for a greeting...</h2>);
+}
+
+export default App;
+```
+
+### Test Event Listener
+
+In the JavaScript console of your browser, enter:
+
+``` js
+const widgetEvent = new CustomEvent('greeting', {
+  detail: {
+    name: 'Pippo'
+  },
+});
+window.dispatchEvent(widgetEvent);
+```
+
+The custom event should now display in the `subscriber-widget`.
+
+::: tip Congratulations!
+You’ve now created a micro frontend that listens to custom events.
+:::
+
+## Add Widgets to App Builder
+
+Now let's test the publisher and subscriber micro frontends in Entando.
```

Comment:
missing word "... let's **add** the publisher..."

Reply:
Updated.